### PR TITLE
Editor Welcome: Add new slide to the Welcome Tour (EN only!)

### LINKED ISSUE #56709: Editor Welcome Tour: Add new slide for List View
**Note**: when picking this one, first check the changes from https://github.com/Automattic/wp-calypso/pull/56347 as mentioned in https://github.com/Automattic/wp-calypso/issues/56709#issuecomment-987785753
___

### Details
We'd like to introduce a new slide to the Editor Welcome Tour to present the List View feature.

https://user-images.githubusercontent.com/12104589/135470978-95e76aad-c565-4aff-89e4-910bbe40b000.mov

### Checklist (Acceptance Criteria)

- [x] Desktop (and mobile) copy @donalirl 
- [x] Desktop (and mobile) graphics @donalirl 
- [ ] Implementation - add this as the 6th slide, only on desktop resolutions
- [ ] Handle the i18n (hide it for non-EN and open a follow-up issue to display it when translations are ready)

### PR DESCRIPTION
**Pull request by StefanNieuwenhuis**
# NOTE:  D72162-code has to be approved and merged before merging this PR!
 
### Changes proposed in this Pull Request

* Add a new slide "Find your way" to the Editor Welcome Tour

#### Technical changes

In `apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-steps.js`:
* Update tour assets config.
* Update Reference Elements config.
* Update Tour Steps config.

### Testing instructions

**Preparations**

1. Make sure that you're sandboxed
2. Apply D72162-code w/ `arc patch-image D72162` (see PCYsg-h9w-p2)
3. Make sure that s0.wp.com is sandboxed
4. Check out this branch and `yarn && yarn dev --sync`.

**Testing instructions**

1. Navigate to one of your sandboxed sites and clear the browser's local storage.
2. Navigate to the editor
3. The welcome tour should pop up. If not, launch it from the more settings menu.
4. Validate that:
     - [ ] All animations work as expected.
     - [ ] The animations for desktop are visible.

**i18n**

image

Fixes #56709

### CODE REVIEW

**Review by escapemanuele — COMMENTED** [review]
Thanks for working on this! 
I see that also this new gif overlaps with the header controls a little, even if in this case is not a bad effect.
We could go on with this and then, if we need, do something[ like this](https://github.com/Automattic/wp-calypso/pull/59462/files#diff-53be941b1ae177b2de28e3f8a62c327950d176d8d357ef0fce56a60764bd8330R219-R227) to style it using CSS! WDYT?

File: packages/tour-kit/src/types.ts [hunk 1]
```diff
@@ -2,10 +2,6 @@ import React from 'react';
 import type { Modifier } from 'react-popper';
 
 export type Step = {
-	referenceElements?: {
```
Comment (chriskmnds) [inline packages/tour-kit/src/types.ts]: Not sure how this is improving things. `meta` (which we may potentially refactor to `content`) being arbitrary is more of a necessary evil to accommodate consumers defining some content structure that fits their renderers. Treating `referenceElements` as part of arbitrary content means we have to assert/infer its presence, when there's nothing really arbitrary about it. Tour kit expects it, it's not a consumer-facing structure. It's also not metadata. It should also be clear to communicate about it.

File: packages/tour-kit/src/storybook/index.stories.tsx [hunk 2]
```diff
@@ -29,39 +31,30 @@ const Tour = ( { onClose, options }: { onClose: () => void; options?: Config[ 'o
...
 				},
 			},
 			{
-				referenceElements: {
-					desktop: '.storybook__tourkit-references-c',
-					mobile: '.storybook__tourkit-references-c',
-				},
 				meta: {
-					description: 'Lorem ipsum dolor sit amet.',
+					description: 'Donec dui sapien, tincidunt eget.',
```
Comment (chriskmnds) [inline packages/tour-kit/src/storybook/index.stories.tsx]: NIT: The reason these were the same was to get an identical step view near all the reference elements in the story. We should either revert or accommodate by setting specific dimensions (height/width) to each step in the storybook. Why is this change relevant to this PR though?

**Review by chriskmnds — CHANGES_REQUESTED** [review]

File: packages/tour-kit/src/types.ts (hunk 1 above)
Comment (escapemanuele) [reply packages/tour-kit/src/types.ts]: Yes, same concerns. I just inherited this PR, and I tried to get it to work after rebasing and everything (also failed to remove the 'Need review' apparently) 🤔  
I asked, and the motivation behind this change was to have it more in order and also remove the `referenceElements` array.

I propose that we keep the `referenceElements` outside of `meta`, but maintain the remotion of the `referenceElements` array, something like this: 

```
{
	referenceElements: referencePositioning && {
		mobile:
			'.edit-post-header .edit-post-header__settings .interface-pinned-items > button:nth-child(1)',
		desktop:
			'.edit-post-header .edit-post-header__settings .interface-pinned-items > button:nth-child(1)',
	},
	meta: {
		heading: __( 'More Options', 'full-site-editing' ),
		descriptions: {
                         ....
```

File: packages/tour-kit/src/storybook/index.stories.tsx (hunk 2 above)
Comment (escapemanuele) [reply packages/tour-kit/src/storybook/index.stories.tsx]: Not sure about that, reverted to how it was before

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-steps.js [hunk 3]
```diff
@@ -31,48 +34,9 @@ function getTourAssets( key ) {
...
-	{
-		desktop: null,
-		mobile: null,
-	},
-];
-
 function getTourSteps( localeSlug, referencePositioning ) {
 	return [
 		{
-			referenceElements: referencePositioning && referenceElements[ 0 ],
```
Comment (escapemanuele) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-steps.js:75]: Anyway, we could decide to leave `referenceElements: null` or remove it alltogether, when no reference element is present

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss [hunk 4]
```diff
@@ -216,13 +216,27 @@ $welcome-tour-card-media-extra-padding: 14%; // temporary value, to match the pa
...
+		.wpcom-editor-welcome-tour__step {
+			&.is-with-extra-padding-desktop {
+				.components-card__media img {
+					left: $welcome-tour-card-media-extra-padding;
+					top: $welcome-tour-card-media-extra-padding;
+					width: 100% - $welcome-tour-card-media-extra-padding;
+				}
 			}
 		}
 	}
```
Comment (escapemanuele) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: Experimenting with a working version.
The problem here is that the new slide is desktop only, and it needs extra padding. Until now, all the images that needed extra padding were only the mobile ones, so the CSS worked.

Random thoughts:
- it'd be difficult for the consumer to apply the right CSS when there are differences between mobile and desktop classes, like in this case. 
- I was thinking about something like:
```js
options: {
		className: {
			desktop: 'is-with-extra-padding',
			mobile: '',
		},
	},
```

This is just a random thought, I should study it a little more, but wanted to know your idea @chriskmnds 

In that case, we could rethink the CSS as:
```css
.wpcom-editor-welcome-tour {
	.wpcom-editor-welcome-tour__step {
		&.is-with-extra-padding {
			.components-card__media img {
				left: $welcome-tour-card-media-extra-padding;
				top: $welcome-tour-card-media-extra-padding;
				width: 100% - $welcome-tour-card-media-extra-padding;
			}
		}
	}
}
```

File: packages/tour-kit/src/types.ts (hunk 1 above)
Comment (chriskmnds) [reply packages/tour-kit/src/types.ts]: Sounds good. They are two separate things. I'm fine with whatever, as long as we avoid touching/modifying Tour Kit types for that purpose here, at this point.

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-steps.js (hunk 3 above)
Comment (chriskmnds) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-steps.js:75]: `referenceElements` should be optional, so we can remove if none needed. This is the type I have on my end right now:

```
export interface Step {
	referenceElements?: {
		desktop?: string;
		mobile?: string;
	};
	meta: {
		[ key: string ]: unknown;
		// | React.FunctionComponent< Record< string, unknown > >
		// | HTMLElement
		// | string
		// | ...
	};
	options?: {
		className?: string;
	};
}
```

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss (hunk 4 above)
Comment (chriskmnds) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: I don't think that's necessary. We provide `is-mobile` & `is-desktop` modifiers on the parent (`wpcom-editor-welcome-tour` in this case), so we should be able to work with those and a single modifier? Else what do we need to provide the modifiers for?

Comment (chriskmnds) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: p.s. happy to add those modifiers to the steps as well if it will ease with usage. But will it? 🤔 

Also, the consumer can choose to conditionally provide modifiers if it wants to right? We just give convenience classes to assist, ~since we (Tour Kit) also control what is "mobile" and what is "desktop"~ (edit: I guess that's the same either way though - we still need to communicate what desktop and mobile are at other places).

Comment (escapemanuele) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: My concern was about creating another class `is-with-extra-padding-desktop`, with the same content, as in the example above!
If we simply use `is-with-extra-padding`, I have to tell to apply that class only on mobile in some slides and only on desktop in others 🤔   For example, in the first slide we don't want the class to be applied to the desktop version.

Well anyway, a simple CSS solution is the one above, it seemed a little confusing tho 🤔 
I may be overcomplicating things and the CSS solution could be much simpler than I think :D

Comment (chriskmnds) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: Up to you on this one! I'd be fine with either way I think. But if we'll have no use for the modifiers added by Tour Kit, we should probably remove them. 

p.s. whatever we choose, let's also communicate in the README the distinction between mobile & desktop at the places where it's mentioned (cause so far it's only deep inside Tour Kit). I meant to add a note about the `is-mobile` and `is-desktop` modifiers, so may as well take care of it now. :-)

Comment (chriskmnds) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: > I'd be fine with either way I think.

but happy to rethink a little more. I'm fiddling with this structure for the Step meta for the variant:

```
export interface WpcomStep extends Step {
	meta: {
		heading: string | null;
		description: {
			desktop: string | null;
			mobile: string | null;
		};
		imgSrc?: {
			desktop?: {
				src: string;
				type: string;
			};
			mobile?: {
				src: string;
				type: string;
			};
		};
	};
}
```

Not final, but we do structure things as `{ mobile | desktop }` a lot, so may as well do it for everything? Feels almost like two separate configurations might work 😵

Comment (chriskmnds) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: > Feels almost like two separate configurations might work 😵

For example, we could have something like `config.step` and `config.stepDesktop` variables (or `config.step.{ default | desktop }`), both having the same structure:

```
interface Step {
    referenceElement?: string;
    options?: {
        classNames?: string[];
    };
    meta: {
        [ key: string ]: unknown;
    };
};

interface WpcomStep extends Step {
    meta: {
        heading?: string;
        description?: string;
        imgSrc?: {
            src: string;
            type: string;
        };
    };
}
```

if both are provided, then Tour Kit picks the right one for each case! I think I'm liking this! will see if can be applicable to the whole config 😆

Comment (escapemanuele) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: I just have the feeling that for a generic customer it'd be easier to just say: 
```css
options: {
  className: {
	  desktop: 'is-with-extra-padding',
	  mobile: null, //in this case no class applied to the step when in mobile
  },
},
```
rather than work with `is-mobile` and `is-desktop` modifiers. And as you said, that mimics how we structured the rest of the configuration 🤔

Comment (escapemanuele) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: We could ship this PR with the CSS I tried, as the main purpose is to add the new slide, and then try the new approach in a follow-up and see if we like it 😃

Comment (escapemanuele) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: Oh didn't saw your last comment, that's not bad at all! I'd explore it in another PR though 😄

Comment (escapemanuele) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: > if both are provided, then Tour Kit picks the right one for each case

I like this a lot to be honest, it seems just a lot easier for the user to use that in some way instead of working with CSS classes. It also should be easier to explain

Comment (chriskmnds) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: > I just have the feeling that for a generic customer it'd be easier to just say:

Ok, let's do that then and get rid of `is-desktop` / `is-mobile`. Let's handle it in another PR if you prefer (mind creating an issue? -> something specific like "Deprecate is-mobile/is-desktop modifiers in favor of granular className definitions".

> Oh didn't saw your last comment, that's not bad at all! I'd explore it in another PR though

Definitely not something for here. 😆  I'm exploring on my end as I'm revising the types with the variant extraction I think it will likely come with some variation of that. There's just too much in the way of `desktop | mobile` everywhere.

Comment (escapemanuele) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: Sure, created issue [60345](https://github.com/Automattic/wp-calypso/issues/60345) 🚀

Comment (escapemanuele) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: Do you think we can ship this PR?

Comment (chriskmnds) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss:242]: After you double-fixed the README, then yes, it should be good now!

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/test/tour-steps.test.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	const originalModule = jest.requireActual( '@wordpress/i18n' );
+	return {
+		__esModule: true,
+		...originalModule,
+		hasTranslation: jest.fn( () => true ),
+	};
+} );
+
+const MOCK_DEFAULT_LOCALE_SLUG = 'en';
+const MOCK_REFERENCE_POSITIONING = true;
```
Comment (chriskmnds) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/test/tour-steps.test.ts]: Maybe we can remove this constant if it's always used in all the tests below.

**Review by chriskmnds — APPROVED** [review]
Do a few more tests please to make sure this is not rushed out. Also make sure the new asset is added properly. :shipit: if it tests out

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7120036

Thank you @StefanNieuwenhuis for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7120036).
